src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +112,872 @@ static DamageTable D_80AD2550 = {
+    EnOsn* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 20.0f);
+    SkelAnime_InitFlex(globalCtx, &this->anime, &D_060202F0, (AnimationHeader*)(&D_060201BC), 0, 0, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->unk_1FC = -1;
+    switch (this->actor.params & 3) {
+        case 0:
+            if (((gSaveContext.entranceIndex == 0xC020) || (gSaveContext.entranceIndex == 0xC030)) ||
+                (gSaveContext.entranceIndex == 0xC060)) {
+                this->unk_1EA = (u16)(this->unk_1EA | 1);
+            }
+            this->unk_1F0 = 1;
+            if (globalCtx->sceneNum == 0x63) {
+                if ((gSaveContext.entranceIndex == 0xC020) || (gSaveContext.entranceIndex == 0xC060)) {
+                    this->actionFunc = func_80AD16A8;
+                    return;
+                }
+                if (gSaveContext.entranceIndex == 0xC030) {
+                    func_80AD1398(this);
+                    this->actionFunc = func_80AD1634;
+                    return;
+                }
+                func_80AD144C(this, globalCtx);
+                return;
+            }
+            func_80AD144C(this, globalCtx);
+            return;
+
+        case 1:
+            this->unk_1EC = 0xF;
+            Actor_ChangeAnimation(&this->anime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            return;
+
+        case 2:
+            this->unk_1EC = 0x10;
+            Actor_ChangeAnimation(&this->anime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            return;
+
+        case 3:
+            this->actor.flags &= -2;
+            this->actionFunc = func_80AD16A8;
+            return;
+
+        default:
+            Actor_MarkForDeath(&this->actor);
+            return;
+    }
+}
+
+void EnOsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsn* this = THIS;
+
+    SkelAnime_Free(&this->anime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnOsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    u32 pad1;
+    u16 pad2;
+    u32 sp34;
+    EnOsn* this = THIS;
+
+    sp34 = Flags_GetSwitch(globalCtx, 0);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    SkelAnime_Update(&this->anime);
+    if ((this->actor.params & 3) == 0) {
+        if (sp34 != 0) {
+            this->actor.flags |= 1;
+            func_80AD0830(this, globalCtx);
+            this->actor.draw = EnOsn_Draw;
+        } else {
+            this->actor.draw = 0;
+            this->actor.flags &= -2;
+        }
+    }
+    func_800E9250(globalCtx, &this->actor, (Vec3s*)this->unk_1D8, (Vec3s*)&this->unk_1D8[6], this->actor.focus.pos);
```

Comment:
This part of the code tells you `unk_1D8` is actually a Vec3s, so you should change it accordingly.
`unk_1D8`+6 (`unk_1DE`) is also a Vec3s, so that should be changed too